packages/components/src/tabs/types.ts

```diff
@@ -18,104 +18,144 @@ export type TabsContextProps =
+	 * `false`, the tab will only be selected upon clicking, not when using arrow
+	 * keys to shift focus (manual tab activation). See the official W3C docs
+	 * for more info.
 	 *
 	 * @default true
 	 *
 	 * @see https://www.w3.org/WAI/ARIA/apg/patterns/tabpanel/
 	 */
-	selectOnMove?: boolean;
+	selectOnMove?: Ariakit.TabStoreProps[ 'selectOnMove' ];
 	/**
-	 * The id of the tab to be selected upon mounting of component.
-	 * If this prop is not set, the first tab will be selected by default.
-	 * The id provided will be internally prefixed with the
-	 * `TabsContextProps.instanceId`.
+	 * The id of the tab whose panel is currently visible.
+	 *
+	 * If left `undefined`, it will be automatically set to the first enabled
+	 * tab, and the component assumes it is being used in "uncontrolled" mode.
+	 *
+	 * Consequently, any value different than `undefined` will set the component
+	 * in "controlled" mode. When in "controlled" mode, the `null` value will
+	 * result in no tabs being selected, and the tablist becoming tabbable.
+	 */
+	selectedTabId?: Ariakit.TabStoreProps[ 'selectedId' ];
+	/**
+	 * The id of the tab whose panel is currently visible.
+	 *
+	 * If left `undefined`, it will be automatically set to the first enabled
+	 * tab. If set to `null`, no tab will be selected, and the tablist will be
+	 * tabbable.
 	 *
 	 * Note: this prop will be overridden by the `selectedTabId` prop if it is
-	 * provided. (Controlled Mode)
+	 * provided (meaning the component will be use in "controlled" mode).
 	 */
-	defaultTabId?: string;
+	defaultTabId?: Ariakit.TabStoreProps[ 'defaultSelectedId' ];
 	/**
-	 * The function called when a tab has been selected.
-	 * It is passed the id of the newly selected tab as an argument.
+	 * The function called when the `selectedTabId` changes.
 	 */
-	onSelect?: ( selectedId: string | null | undefined ) => void;
-
+	onSelect?: Ariakit.TabStoreProps[ 'setSelectedId' ];
 	/**
-	 * The orientation of the tablist.
+	 * The current active tab `id`. The active tab is the tab element within the
+	 * tablist widget that has DOM focus.
+	 * - `null` represents the tablist (ie. the base composite element). Users
+	 *   will be able to navigate out of it using arrow keys.
+	 * - If `activeTabId` is initially set to `null`, the base composite element
+	 *   itself will have focus and users will be able to navigate to it using
+	 *   arrow keys.activeTabId
+	 */
+	activeTabId?: Ariakit.TabStoreProps[ 'activeId' ];
+	/**
+	 * The tab id that should be active by default when the composite widget is
+	 * rendered. If `null`, the tablist element itself will have focus
+	 * and users will be able to navigate to it using arrow keys. If `undefined`,
+	 * the first enabled item will be focused.
 	 *
-	 * @default `horizontal`
+	 * Note: this prop will be overridden by the `activeTabId` prop if it is
+	 * provided.
+	 */
+	defaultActiveTabId?: Ariakit.CompositeStoreProps[ 'defaultActiveId' ];
```

Comment:
Any good reason why we're using `CompositeStoreProps` and not `TabStoreProps` here?

Reply:
I can think of a couple of reasons, but none of them are "good" 🤦  Just tiredness, i guess. I will updated